ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/user_management_service.proto

```diff
@@ -18,6 +21,16 @@ option csharp_namespace = "Com.Daml.Ledger.Api.V1.Admin";
 // (1) ``HasRight(r)`` denoting whether the authenticated user has right ``r`` and
 // (2) ``IsAuthenticatedUser(uid)`` denoting whether ``uid`` is the empty string or equal to the id of the authenticated user.
 //
+// We annotate the fields of request messages (and their sub-messages) with 'Optional' and 'Required' markers.
+// 'Optional' means that the client can leave the field unset (set to a default value) when sending a request.
+// 'Required' means that the client must set the field to a non-default value when sending a request.
+//
+// For request messages (and their sub-messages) used in 'Update<Resource>' calls we annotate their fields, corresponding to the resource being update,
+// with 'Read-Only' and 'Read-Write' markers.
+// 'Read-Only' means that the field (and its sub-fields) cannot be updated; i.e. the field's name can never be used in a valid update mask's path.
```

Comment:
'Read-Only' might be misleading. It could suggest that the value cannot be set by the client (like with 'is_local' property of a party).
That's not true for user id - client is responsible for choosing and setting it. It is however unmodifiable once a user has been created.

Instead of using 'Read-Only' and 'Read-Write', we could instead mark appropriate fields as 'Modifiable' or 'Updatable'.

Reply:
Good point. I like 'Modifiable'.